### PR TITLE
[ASA-659] Connect to DES rather than squid for NPS employments

### PR DESCRIPTION
**Pull request by geoffreywatson**

### CODE REVIEW

File: app/uk/gov/hmrc/taxhistory/metrics/MetricsEnum.scala [hunk 1]
```diff
@@ -25,4 +25,5 @@ object MetricsEnum extends Enumeration {
   val NPS_GET_IABDS = Value
   val NPS_GET_TAX_ACCOUNT = Value
   val CITIZEN_DETAILS = Value
+  val DES_NPS_GET_EMPLOYMENTS = Value
```
Comment (johnrevill) [inline app/uk/gov/hmrc/taxhistory/metrics/MetricsEnum.scala]: We've already got `NPS_GET_EMPLOYMENTS` - is this the same thing?

File: test/uk/gov/hmrc/taxhistory/connectors/DesNpsConnectorSpec.scala [hunk 2]
```diff
@@ -173,6 +179,53 @@ class DesNpsConnectorSpec extends PlaySpec with MockitoSugar with TestUtil {
...
+          .thenReturn(Future.failed(new Upstream5xxResponse("", INTERNAL_SERVER_ERROR, INTERNAL_SERVER_ERROR)))
+          .thenReturn(Future.successful(testNpsEmployment))
+
+        val result = testemploymentsConnector.getEmployments(testNino, testYear)
+
+        await(result) mustBe testNpsEmployment
+      }
+
+      "return and handle an error response" in {
+        val expectedResponse = Json.parse( """{"reason": "Some thing went wrong"}""")
```
Comment (johnrevill) [inline test/uk/gov/hmrc/taxhistory/connectors/DesNpsConnectorSpec.scala]: `expectedResponse` is unused

File: test/uk/gov/hmrc/taxhistory/services/EmploymentHistoryServiceSpec.scala [hunk 3]
```diff
@@ -307,13 +308,20 @@ class EmploymentHistoryServiceSpec extends UnitSpec with MockitoSugar with TestU
...
-      intercept[NotFoundException](await(testEmploymentHistoryService.getTaxAccount(testNino, TaxYear.current.previous)))
-    }
+        intercept[NotFoundException](await(testEmploymentHistoryService.getTaxAccount(testNino, TaxYear.current.previous)))
+      }
 
     "fail with a NotFoundException if asked for TaxAccount for the current year" in new StubConnectors {
       intercept[NotFoundException](await(testEmploymentHistoryService.getTaxAccount(testNino, TaxYear.current)))
     }
 
+    "return None from get Nps Tax Account api for not found response " in {
```
Comment (johnrevill) [inline test/uk/gov/hmrc/taxhistory/services/EmploymentHistoryServiceSpec.scala]: Test name doesn't match with what the test is doing

File: app/uk/gov/hmrc/taxhistory/metrics/MetricsEnum.scala (hunk 1 above)
Comment (geoffreywatson) [reply app/uk/gov/hmrc/taxhistory/metrics/MetricsEnum.scala]: This is to differentiate between squid and des calls for get employments. If not needed then use the `NPS_GET_EMPLOYMENTS`

Comment (johnrevill) [reply app/uk/gov/hmrc/taxhistory/metrics/MetricsEnum.scala]: These metrics end up in the grafana dashboard (https://grafana.tools.production.tax.service.gov.uk/d/LvOXXDOmz/tax-history?orgId=1) as just "NPS Get Employment Call Count" and "NPS Get Employment Response Time".
If we use a different metric name, we'd need to change the Grafana dashboard configuration too. 
We wont need to distinguish between Squid and DES in the metrics though, so we may as well keep the same previous name of `NPS_GET_EMPLOYMENTS`.

**Pull request merged**